CRM/Core/Payment.php

```diff
@@ -1836,4 +1835,13 @@ public function isSendReceiptForPending() {
     return FALSE;
   }
 
+  /**
+   * Get the result array to return from doPayment on success.
+   *
+   * @return array
+   */
+  protected function getSuccessResult(): array {
+    return ['payment_status_id' => CRM_Core_PseudoConstant::getKey('CRM_Contribute_BAO_Contribution', 'contribution_status_id', 'Completed')];
+  }
```

Comment:
@eileenmcnaughton Not sure I see the value of splitting this out for just one line of code?  If it *is* split out the function name should be closely tied to doPayment so it doesn't get repurposed for things it shouldn't.

Reply:
@mattwire the reasons I thought to do it were
1) although it's just 1 line it's a pretty painful line that is inconsistently done in payment processors
2) I wanted to propose also adding a key 'payment_status' => 'Completed' and migrating over time to that because it's always seemed like doing an option group lookup is a long way around to me.